### PR TITLE
executor: rename txnScope as readReplicaScope instead

### PR DESCRIPTION
**Pull request by Yisaer**
### What is changed and how it works?

txnScope generally indicates `tso scope` and `request store scope`. In exector, the txnScope only indicates request scope so that I rename it as `readReplicaScope` in order to avoid misleading meaning.

### Release note 

```release-note
None
```

### CODE REVIEW

File: distsql/request_builder_test.go [hunk 1]
```diff
@@ -252,17 +252,17 @@ func TestRequestBuilder1(t *testing.T) {
...
+		KeepOrder:        false,
+		Desc:             false,
+		Concurrency:      variable.DefDistSQLScanConcurrency,
+		IsolationLevel:   0,
+		Priority:         0,
+		NotFillCache:     false,
+		SyncLog:          false,
+		Streaming:        false,
+		ReplicaRead:      kv.ReplicaReadLeader,
+		ReadReplicaScope: oracle.GlobalTxnScope,
```
Comment (nolouch) [inline distsql/request_builder_test.go:265]: IMO，`ReplicaScope` and `TxnScope` sound confusing, maybe `ReadReplicaScope: Globally/Locally` more suitable. or add some comments.

Comment (Yisaer) [reply distsql/request_builder_test.go]: updated.

**Review by JmPotato — APPROVED** [review]

**Review by nolouch — APPROVED** [review]

**Review by disksing — APPROVED** [review]

**Pull request merged**